src/controller/eme-controller.ts

```diff
@@ -890,27 +895,66 @@ class EMEController extends Logger implements ComponentAPI {
           keyId = status;
           status = temp;
         }
-        const keyIdWithStatusChange = arrayToHex(
-          'buffer' in keyId
-            ? new Uint8Array(keyId.buffer, keyId.byteOffset, keyId.byteLength)
-            : new Uint8Array(keyId),
-        );
-
+        
+        let keyIdArray: Uint8Array = 'buffer' in keyId
+          ? new Uint8Array(keyId.buffer, keyId.byteOffset, keyId.byteLength)
+          : new Uint8Array(keyId);
+        
+        // Handle PlayReady little-endian key ID conversion for status comparison only
+        // Don't modify the original key ID from playlist parsing
+        if (mediaKeySessionContext.keySystem === KeySystems.PLAYREADY && keyIdArray.length === 16) {
+          keyIdArray = new Uint8Array(keyIdArray);
+          changeEndianness(keyIdArray);
+        }
+        
+        const keyIdWithStatusChange = arrayToHex(new Uint8Array(keyIdArray));
+        
+        // Store all key statuses for processing
+        keyStatuses.push({ status, keyId: keyIdWithStatusChange });
+        
         // Error immediately when encountering a key ID with this status again
         if (status === 'internal-error') {
           this.bannedKeyIds[keyIdWithStatusChange] = status;
         }
 
-        // Only acknowledge status changes for level-key ID
+        // Check if this key matches the session-level key ID
         const matched = keyIdWithStatusChange === sessionLevelKeyId;
-        this.log(
-          `${matched ? '' : 'un'}matched key status change "${status}" for keyStatuses keyId: ${keyIdWithStatusChange} session keyId: ${sessionLevelKeyId} uri: ${mediaKeySessionContext.decryptdata.uri}`,
-        );
         if (matched) {
+          hasMatchedKey = true;
           mediaKeySessionContext.keyStatus = status;
+          this.log(
+            `matched key status change "${status}" for keyStatuses keyId: ${keyIdWithStatusChange} session keyId: ${sessionLevelKeyId} uri: ${mediaKeySessionContext.decryptdata.uri}`,
+          );
+        } else {
+          this.log(
+            `unmatched key status change "${status}" for keyStatuses keyId: ${keyIdWithStatusChange} session keyId: ${sessionLevelKeyId} uri: ${mediaKeySessionContext.decryptdata.uri}`,
+          );
         }
       },
     );
+    
+    // Handle case where no keys matched but all have the same status
+    // This can happen with PlayReady when key IDs don't align properly
+    if (!hasMatchedKey && keyStatuses.length > 0) {
+      const firstStatus = keyStatuses[0].status;
+      const allSameStatus = keyStatuses.every(({ status }) => status === firstStatus);
```

Comment:
The library has avoided the use of `Array.prototype.every`, I think mostly because of Android browser (pre 4.4.4)  support. Sorry the linter did not catch this. I'll add it to the eslint rules.

For now:
```suggestion
      const allSameStatus = !keyStatuses.some(({ status }) => status !== firstStatus);
```